pkg/storegateway/indexheader/binary_reader.go

```diff
@@ -363,10 +367,10 @@ func (fw *FileWriter) Remove() error {
 	return os.Remove(fw.name)
 }
 
-func (w *binaryWriter) AddIndexMeta(indexVersion int, indexPostingOffsetTable uint64) error {
+func (w *binaryWriter) AddIndexMeta(indexVersion int, indexLastPostingListEndBound uint64) error {
 	w.buf.Reset()
 	w.buf.PutByte(byte(indexVersion))
-	w.buf.PutBE64(indexPostingOffsetTable)
+	w.buf.PutBE64(indexLastPostingListEndBound)
```

Comment:
[nit] I would add a comment here, saying that in Mimir 2.7 and before this was XXX while now it's YYY.